# HCDE 410 Final Project: Trying to Understand Drug Related overdoses in Connecticut
## By: Isha Narayanan

## Part 1: Loading the dataset
Before dividing into the dataanalytics involed in this project, I first worked to load the complete dataset into the drug_addiction variable. From my initial findings, there are 9202 rows in this dataset. 

In [2]:
import csv

def prepare_datasets(file_path):
    """ 
    Accepts: path to a tab-separated plaintext file
    Returns: a list containing a dictionary for every row in the file, 
        with the file column headers as keys
    """
    
    with open(file_path) as infile:
        reader = csv.DictReader(infile, delimiter=',')
        list_of_dicts = [dict(r) for r in reader]
        
    return list_of_dicts

In [3]:
drug_addiction = prepare_datasets("Accidental_Drug_Related_Deaths_2012-2021.csv")


In [4]:
print(len(drug_addiction))
print(drug_addiction[0])

9202
{'Date': '05/29/2012', 'Date Type': 'Date of death', 'Age': '37', 'Sex': 'Male', 'Race': 'Black', 'Ethnicity': '', 'Residence City': 'STAMFORD', 'Residence County': 'FAIRFIELD', 'Residence State': '', 'Injury City': 'STAMFORD', 'Injury County': '', 'Injury State': 'CT', 'Injury Place': 'Residence', 'Description of Injury': 'Used Cocaine', 'Death City': '', 'Death County': '', 'Death State': '', 'Location': 'Residence', 'Location if Other': '', 'Cause of Death': 'Cocaine Toxicity', 'Manner of Death': 'Accident', 'Other Significant Conditions ': '', 'Heroin': '', 'Heroin death certificate (DC)': '', 'Cocaine': 'Y', 'Fentanyl': '', 'Fentanyl Analogue': '', 'Oxycodone': '', 'Oxymorphone': '', 'Ethanol': '', 'Hydrocodone': '', 'Benzodiazepine': '', 'Methadone': '', 'Meth/Amphetamine': '', 'Amphet': '', 'Tramad': '', 'Hydromorphone': '', 'Morphine (Not Heroin)': '', 'Xylazine': '', 'Gabapentin': '', 'Opiate NOS': '', 'Heroin/Morph/Codeine': '', 'Other Opioid': '', 'Any Opioid': '', 'Oth

## Part 2: Introduction 
### High Level Overview: 
I am planning to do this analysis because the United States of America is number one in the world for drug related deaths. Furthemore, drug related deaths are projected to increase. Lastly, drug addiction and drug related deaths have exploded in Seattle over the past 5 years. I think its important to understand the potnetial correlation between intersections of identity and addiction and drug related deaths. Classism and systemic racism are deep rooted in America and I want to understand if these factors tangibly affect drug related deaths. I want to understand the if there is a strong relationship between the following variables and Drug related deaths: Age, Race, and Sex. Furthermore, I hope do in depth analysis of the ratios of specific drugs used in overdoses. I hope to learn how these factors affect drug related deaths. Afterwards I am going to do more research to understand why. 
### Data Selected for Analysis:
- Link to Data Set: https://www.kaggle.com/datasets/utkarshx27/accidental-drug-related-deaths-2012-2021
- License: U.S. Government Works https://www.usa.gov/government-works/
This dataset analyzes accidental deaths associated with drug overdoses in Connecticut from 2012 to 2021. The data is derived from an investigation by the Office of the Cheif Medical Examiner which includes the toxicity report and death certificate. There are 48 columns in the dataset and 9202 rows. This dataset is suitable for addressing my research goal because it contains all of the variables that I wanted to analyze and more (there are 48 columns). Furthermore, the dataset is very large and spans from 2012-2021. One possible ethical issue with this data set is that the data set may not have had the permission of the deceased person or the decreased person's family to publish their data in this way. Secondly, alot of drug related deaths go unreported so conclusions that are reached through this data may not be entirely accurate. 

## Part 3: Background or Related Work 
- There has been alot of previous research into drug addiction in the United States. This is evidenced through a quick google search of the topic. I was able to find that more than 106,000 people in the U.S. died from drug-involved overdoses in 2021; this is a 781% increase since 1999 (https://nida.nih.gov/research-topics/trends-statistics/overdose-death-rates#:~:text=More%20than%20106%2C000%20persons%20in,drugs%20from%201999%20to%202021.). Pertaining to drug related over doses from all drugs; Washington DC, West Virginia, and Kentucky have the highest rates in that order. Furthermore, more than 75% of drug overdoses involve opiods. 
- According to a study conducted by Pew Research Center published on January 19, 2022 (https://www.pewresearch.org/short-reads/2022/01/19/recent-surge-in-u-s-drug-overdose-deaths-has-hit-black-men-the-hardest/), black men have been hit the hardest by the U.S. drug overdose death surge. Since 2015, the death rate among black men has tripping while the rates of death among other races have increased at a much slower pace. The study also takes about the death rates of other races(White, American Indian, Alaskan Native). Diving into this study lead me to really want to understand how and why race or ethnicty plays such an important role in the drug addiction or drug overdoses today. Particularly, how have these dynamics changes/not changed in recent years. The long timeline of my chosen data set (2012-2021) will allow me to compute and compare these metrics accross different time periods. Furthermore, this study talks about the sharp rise in drug realted deaths amongst women(specifically black women) in the United States. This also peaks my interest in the relation ship between sex and drug realted deaths and how that relationship has changed over time. 

## Part 4: Research Questions 
I believe that these two research questions will help me understand this issue from the perspective of the drugs themselves and also identity: 
- How do intersections of identity affect drug overdose rates(Biological Sex and Race)? Why is this the case? 
- What are the drugs that are most common in drug overdoses in Connecticut? Why is this the case? 
 

## Part 5: Methodology
I intend on conducting four different pieces of analysis on different variables. These variables in the dataset include Age, Sex, and Race. Furthermore, I will also anlyze: Heroin, Cocaine, Fentanyl, Fentanyl Analouge, Oxycodone, Oxymorphone, Ethanol, Hydrocodone, Benzodiazepine, Methadone, Meth/Amphetamine, Amphet, Tramad, Hydromorphone, Morphine(Not Herion), Xylazine, Gabapentin, Opiate NOS, Other Opioid, and Any Opioid. 

- For the following variables, I will be using the same analysis process: Age, Sex, Race
    - 1) Get the total number of unique values in the column 
    - 2) Calculate how many rows are missing 
    - 3) Calculate percentages of each unique value in the column 
    - 4) Compare and contrast findings to the general demographic statistics of Connecticut
    - 5) Do supplemental research to contextualize these findings 
- I will also be conducting anlysis on which drugs are most common in drug overdoses: 
    - 1) Create a list for each drug variable and load all non null rows into the respective dictionaries 
    - 2) Print the length of each dictionary and determine which drugs are most common in drug overdoses in Connecticut
    - 3) Do supplemental research to contextualize findings 


### Analyzing the most prevalent drugs in overdoses 
In order to accomplish this goal, I first create a list for each drug variable and count the rows in each list that are not null. Then, I will compare the counts of each drug to each other. From this initial research I can conclude that opiates have been involved in the maximum amount of drug overdoses in Connecticut (6,482). 

In [5]:
import statistics

heroin_list = []

for x in drug_addiction:
    heroin_list.append(x['Heroin'])

countHeroin = 0; 

for y in heroin_list: 
    if(y == 'Y'):
        countHeroin = countHeroin + 1
print("Heroin")
print(countHeroin)
print()

###################

Cocaine_list = []

for x in drug_addiction:
    Cocaine_list.append(x['Cocaine'])
    

countCocaine = 0; 

for y in Cocaine_list: 
    if(y == 'Y'):
        countCocaine = countCocaine + 1
print("Cocaine")
print(countCocaine)
print()

###################

FentanylFentanylAnalouge_list = []

for x in drug_addiction:
    FentanylFentanylAnalouge_list.append(x['Fentanyl'])
    FentanylFentanylAnalouge_list.append(x['Fentanyl Analogue'])
    
countFentanyl = 0; 

for y in FentanylFentanylAnalouge_list: 
    if(y == 'Y'):
        countFentanyl = countFentanyl + 1
print("Fentanyl + Fentanyl Analouge")
print(countFentanyl)
print()

###################

Oxycodone_list = []

for x in drug_addiction:
    Oxycodone_list.append(x['Oxycodone'])
    
countOxycodone = 0; 

for y in Oxycodone_list: 
    if(y == 'Y'):
        countOxycodone = countOxycodone + 1
print("Oxycodone")
print(countOxycodone)
print()

###################

Oxymorphone_list = []

for x in drug_addiction:
    Oxymorphone_list.append(x['Oxymorphone'])
    
countOxymorphone = 0; 

for y in Oxycodone_list: 
    if(y == 'Y'):
        countOxymorphone = countOxymorphone + 1
print("Oxymorphone")
print(countOxymorphone)
print()

###################

Ethanol_list = []

for x in drug_addiction:
    Ethanol_list.append(x['Ethanol'])
    
countEthanol = 0; 

for y in Ethanol_list: 
    if(y == 'Y'):
        countEthanol = countEthanol + 1
print("Ethanol")
print(countEthanol)
print()

###################

Hydrocodone_list = []

for x in drug_addiction:
    Hydrocodone_list.append(x['Hydrocodone'])
    
countHydrocodone = 0; 

for y in Hydrocodone_list: 
    if(y == 'Y'):
        countHydrocodone = countHydrocodone + 1
print("Hydrocodone")
print(countHydrocodone)
print()

###################

Benzodiazepine_list = []

for x in drug_addiction:
    Benzodiazepine_list.append(x['Benzodiazepine'])
    
countBenzodiazepine = 0; 

for y in Benzodiazepine_list: 
    if(y == 'Y'):
        countBenzodiazepine = countBenzodiazepine + 1
print("Benzodiazepine")
print(countBenzodiazepine)
print()

###################

Methadone_list = []

for x in drug_addiction:
    Methadone_list.append(x['Methadone'])
    
countMethadone = 0; 

for y in Methadone_list: 
    if(y == 'Y'):
        countMethadone = countMethadone + 1
print("Methadone")
print(countMethadone)
print()

###################

MethAmphetamine_list = []

for x in drug_addiction:
    MethAmphetamine_list.append(x['Meth/Amphetamine'])
    
countMeth = 0; 

for y in MethAmphetamine_list: 
    if(y == 'Y'):
        countMeth = countMeth + 1
print("Meth/Amphetamine")
print(countMeth)
print()

###################

Amphet_list = []

for x in drug_addiction:
    Amphet_list.append(x['Amphet'])
    
countAmphet = 0; 

for y in Amphet_list: 
    if(y == 'Y'):
        countAmphet = countAmphet + 1
print("Amphet")
print(countAmphet)
print()

###################

Tramad_list = []

for x in drug_addiction:
    Tramad_list.append(x['Tramad'])
    
countTramad = 0; 

for y in Tramad_list: 
    if(y == 'Y'):
        countTramad = countTramad + 1
print("Tramad")
print(countTramad)
print()

###################

Hydromorphone_list = []

for x in drug_addiction:
    Hydromorphone_list.append(x['Hydromorphone'])
    
countHydromorphone = 0; 

for y in Hydromorphone_list: 
    if(y == 'Y'):
        countHydromorphone = countHydromorphone + 1
print("Hydromorphone")
print(countHydromorphone)
print()

###################

Morphine_list = []

for x in drug_addiction:
    Morphine_list.append(x['Morphine (Not Heroin)'])
    
countMorphine = 0; 

for y in Morphine_list: 
    if(y == 'Y'):
        countMorphine = countMorphine + 1
print("Morphine")
print(countMorphine)
print()

###################

Xylazine_list = []

for x in drug_addiction:
    Xylazine_list.append(x['Xylazine'])
    
countXylazine = 0; 

for y in Xylazine_list: 
    if(y == 'Y'):
        countXylazine = countXylazine + 1
print("Xylazine")
print(countXylazine)
print()

###################

Gabapentin_list = []

for x in drug_addiction:
    Gabapentin_list.append(x['Gabapentin'])
    
countGabapentin = 0; 

for y in Gabapentin_list: 
    if(y == 'Y'):
        countGabapentin = countGabapentin + 1
print("Gabapentin")
print(countGabapentin)
print()

###################

Opiate_list = []

for x in drug_addiction:
    Opiate_list.append(x['Opiate NOS'])
    Opiate_list.append(x['Other Opioid'])
    Opiate_list.append(x['Any Opioid'])
    
countOpiate = 0; 

for y in Opiate_list: 
    if(y == 'Y'):
        countOpiate = countOpiate + 1
print("Opiate")
print(countOpiate)
print()

###################

Heroin
3347

Cocaine
3171

Fentanyl + Fentanyl Analouge
6398

Oxycodone
877

Oxymorphone
877

Ethanol
2469

Hydrocodone
155

Benzodiazepine
2238

Methadone
814

Meth/Amphetamine
127

Amphet
285

Tramad
258

Hydromorphone
56

Morphine
46

Xylazine
441

Gabapentin
182

Opiate
6482



### Analyzing Sex and its correlation with Drug Overdoses 
In the following code block, I calulate the percentage of biological Males and Females that have died of drug overdoses in the state of Connecticut from 2012-2021. 

In [12]:
biologicalsex_list = []

for x in drug_addiction:
    biologicalsex_list.append(x['Sex'])

countFemale = 0; 
countMale = 0; 
total = 0;

for y in biologicalsex_list: 
    total = total + 1
    if(y == 'Female'):
        countFemale = countFemale + 1
    if(y == 'Male'):
       countMale = countMale + 1 

print("Male Proportion")
males = (countMale/total)*100
print(males)
       
print("Female Proportion")
females = (countFemale/total)*100
print(females)

print(len(biologicalsex_list))

Male Proportion
74.1795261899587
Female Proportion
25.733536187785266
9202


### Analyzing Race and its correlation with Drug Overdoses 
In the following code block I start off by gathering the unique values in the Race column in the dataset. After this point I calculated the percentage out of total dataset of each race.

In [25]:
Race_list = []

for x in drug_addiction:
    Race_list.append(x['Race'])

unique_race_list = []

for y in Race_list:
    if(unique_race_list.count(y) == 0):
        unique_race_list.append(y)

print(unique_race_list)
print(len(unique_race_list))

print(len(Race_list))

print("_____")

BlackCount = 0
WhiteCount = 0
OtherCount = 0
AsianIndianCount = 0
IndigenousCount = 0
AsianOtherCount = 0
HawaiinCount = 0
IndigenousOther = 0
ChineseCount = 0
KoreanCount = 0

for y in Race_list:
    if(y == 'Black' or y == 'Black or African American'):
        BlackCount = BlackCount + 1
    elif(y == 'White'):
        WhiteCount = WhiteCount + 1
    elif(y == 'Other' or y == '' or y == 'Unknown'):
        OtherCount = OtherCount + 1
    elif(y == 'Asian Indian'):
        AsianIndianCount = AsianIndianCount + 1
    elif(y == 'American Indian or Alaska Native'):
        IndigenousCount = IndigenousCount + 1
    elif(y == 'Asian, Other' or y == 'Other Asian'):
        AsianOtherCount = AsianOtherCount + 1
    elif(y == 'Hawaiian'):
        HawaiinCount = HawaiinCount + 1
    elif(y == 'Native American, Other'):
        IndigenousOther = IndigenousOther + 1
    elif(y == 'Chinese'):
        ChineseCount = ChineseCount + 1
    elif(y == 'Korean'):
        KoreanCount = KoreanCount + 1
        
print((BlackCount/9202) *100)
print((WhiteCount/9202) *100)
print((OtherCount/9202) *100)
print((AsianIndianCount/9202) *100)
print((IndigenousCount/9202) *100)
print((AsianOtherCount/9202) *100)
print((HawaiinCount/9202) *100)
print((IndigenousOther/9202) *100)
print((ChineseCount/9202) *100)
print((KoreanCount/9202) *100)      

['Black', 'White', '', 'Other', 'Unknown', 'Asian Indian', 'Black or African American', 'American Indian or Alaska Native', 'Asian, Other', 'Other Asian', 'Hawaiian', 'Native American, Other', 'Chinese', 'Korean']
14
9202
_____
11.877852640730275
86.4051293197131
1.0975874809823951
0.22821125842208217
0.010867202782003912
0.32601608346011735
0.010867202782003912
0.010867202782003912
0.021734405564007825
0.010867202782003912


## Part 6: Findings 
- Analyzing the most prevalent drugs in overdoses: I found that the most people have died from overdosing on opiates (6482 people). This is consistent with my previous research on overdoses in the United States of America. The following drugs are ordered by prevalence in drug overdoses in Connecticut: 	Fentanyl and Fentanyl Analouge (6398 people), Heroin (3347 people), and Cocaine (3171 people). 

- Analyzing Sex and its correlation with Drug Overdoses: I found that there was biological sex recording for every single individual in this dataset(9202). Furthermore, the proportion of male to female individuals who died from drug overdoses was staggering. 74.18% of individuals who died from drug overdoses in this data set were male and the other 25.73% were female. This is consistent with previous research that I did on drug addiction in the United States; this research showed my that there is a much higher percentage of males that are addicted to drugs than women. 

- Analyzing Race and its correlation with Drug Overdoses: I found that there were 14 accounted for races in this dataset. However, I did notice that there was no category for latin people. The races included seemed to skew towards including multiple types of Asian races: ['Black', 'White', '', 'Other', 'Unknown', 'Asian Indian', 'Black or African American', 'American Indian or Alaska Native', 'Asian, Other', 'Other Asian', 'Hawaiian', 'Native American, Other', 'Chinese', 'Korean']. The highest race percentage of people who died from drug overdoses in Connecticut were white: 86.41%; black people also made up 11.88% of the dataset. According to the US Census Bureau(https://www.census.gov/quickfacts/CT) White people make up 78.8% of the population and Black or African American people make up 12.7%.

## Part 7: Discussion
- I think that the biggest limitation of this project was that a large proportion of drug overdoses are nonfatal. Those overdoses are not being addressed in this dataset and furthermore, according to the Texas Tribune (https://www.texastribune.org/2022/07/05/university-texas-drug-overdose-data/), 50% of non fatal drug overdoses go unreported. Furthermore, my report only considers deaths that occured in Connecticut which may not be representative of the entire United States of America. 
- I think my biggest takeaway from this project has been surrounding the types of drugs that people have overdosed most on in Connecticut. I think this is a place in which something really actionable can be done. From my analysis it was clear that opiates were the major contributor to drug related deaths in Connecticut. I think that its import that community leaders and local governments understand why this is the case. Initiative should be taken to understand why so many people turn to opiates in their community and what can be done to stop this. I think that a great way to do this would be to conduct semi formal studies with survivors of opiate addiction in the local area. 

## Part 8: Conclusion 

Overall, this project was very interesting and fulfilling to complete. I learned alot an issue that is a pressing issue in the United States of America. Furthermore, I was able to understand what drugs contribute most to this epidemic while also analyzing verious other intersections of indentity and drug addiction. 